.github/workflows/code_test.yaml

```diff
@@ -0,0 +1,70 @@
+---
+name: Code test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  integration-tests:
+    name: Run integration tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Install rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      - name: Run integration tests
+        run: cargo test --tests
+
+      - name: Install cargo-tarpaulin
+        run: cargo install cargo-tarpaulin
```

Comment:
Add to `Cargo.toml`?

Reply:
aah yes, didn't touch that as it wan't merged.

Reply:
I think since it is a CLI tool it needs to be installed manually outside of the `Cargo.toml`, or maybe you can add it to the rust-toolchain file

Reply:
Really? What's the difference? Aren't the deps in `Cargo.toml` installed via `cargo install`?